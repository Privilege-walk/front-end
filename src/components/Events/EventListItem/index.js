import React, { useState } from "react";
import { 
    Badge
 } from 'reactstrap';
 import { useNavigate } from "react-router-dom";
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
 import CardActions from '@mui/material/CardActions';


export default function EventListItem({ id, name, status }) {
    const navigate = useNavigate();

    function redirectToQuestions() {
        navigate('/questions', { state: {id, name, status} });
    }

    return (
        <Card className="">
            <CardContent>
            <Typography>{name} <Badge color="primary">{status}</Badge></Typography>
            
            </CardContent>
            <CardActions>
                <Button 
                    variant="outlined" 
                    size="md"
                    onClick={redirectToQuestions}
                >
                    Edit
                </Button>
            </CardActions>
            
        </Card>
    );
}