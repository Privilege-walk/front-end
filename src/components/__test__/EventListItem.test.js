import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import EventListItem from '../events/EventListItem';

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    Navigate: () => "Navigate",
    useNavigate: () => mockedUsedNavigate,
}))

afterEach(jest.clearAllMocks);

test('renders without crashing',  () => {
    let utils;
    act (() => {
        utils = render( <EventListItem name="" status="created" />);
    }); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})

test('renders questions page', async () => {
    render( <EventListItem name="" status="created" />);
    const editBtn = screen.getByText("Edit", {selector: "button"});
    await act(async () => {   
        await fireEvent.click(editBtn);
    });
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
})

test('renders host walk page', async () => {
    render( <EventListItem name="" status="created" />);
    const goLiveBtn = screen.getByText("Go Live", {selector: "button"});
    await act(async () => {   
        await fireEvent.click(goLiveBtn);
    });
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
})