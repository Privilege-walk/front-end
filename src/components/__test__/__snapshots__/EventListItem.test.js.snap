// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
         
        <span
          class="badge bg-primary"
        >
          created
        </span>
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMd MuiButton-outlinedSizeMd MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        id="edit-undefined"
        tabindex="0"
        type="button"
      >
        Edit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMd MuiButton-outlinedSizeMd MuiButtonBase-root  css-hmexeu-MuiButtonBase-root-MuiButton-root"
        id="go-live-undefined"
        tabindex="0"
        type="button"
      >
        Go Live
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
